Update deployment script to deploy to GH pages

ijdickinson opened the pull request:
This commit addresses issue #1, by:

- pulling latest changes from @fruffers repo
- update the build script to push `dist` to the `gh-pages` branch
- update README and CHANGELOG
- removed `packge-lock.json` (obsolete)

ijdickinson committed:
Update deployment script to deploy to GH pages

This commit addresses issue #1, by:

- pulling latest changes from @fruffers repo
- update the build script to push `dist` to the `gh-pages` branch
- update README and CHANGELOG
- removed `packge-lock.json` (obsolete)

joescottdave commented on README.md:
```diff
@@ -48,3 +54,13 @@ yarn lint
+To update the deployment, ensure you don't have any uncommitted changes then run
+the `./deploy.sh` Bash script.
+
+**Note**: this script does do some force checkouts to update `gh-pages` branch, so
```
Is it possible to be alerted to uncommitted changes and abort?

ijdickinson replied:
> Is it possible to be alerted to uncommitted changes and abort?

Good suggestion. Will investigate

joescottdave reviewed (approved):
Looks good

ijdickinson committed:
Guard against uncommitted changes

Don't allow users to deploy if there are uncommitted changes, as the
force checkouts of the gh-pages branch may mean that users lose those
changes.
